sauna-app/pages/typeDefinitions.tsx

```diff
@@ -1,4 +1,4 @@
-interface SingleSauna {
+export interface SingleSauna {
```

Comment:
For solving the TypeScript error with `_id`, here's one possible solution (note: Naming of the types is not the best, I'm only on my first cup of coffee this morning 😅 ): 

Have a common type for all the common properties, and then subtype for the Sauna used in the app, and Sauna from the API:

```ts
interface SaunaProperties {
  // common props
}

interface SaunaResp extends SaunaProperties {
  _id: string
}

interface SaunaInApp extends SaunaProperties {
  id: string
}

type SingleSauna = SaunaInApp | SaunaResp
```

This way, either `id` or `_id` is mandatory property (but not both).

Reply:
I was struggling with this a lot and somehow I could not make it work, I now used maybe a bad solution to use only _id (instead of id) everywhere to get rid of the warnings, I don't know is this a bad way to do it or does it really matter how I have named them (_id/id) ?

Reply:
I don't know if it's a bad one - it's a solution. There might be some people who say that it's unconventional, but then again, I think everything is just about agreements and how we set the conventions. Some use underscore to annotate private fields (esp. with classes). 

So I'm okay to accept this solution, just a note that when you're interviewing with some company, they might ask about this so be prepared to explain why you ended up with this solution. (My answer would be something in the lines of "I was lazy and wanted to use the same interface for the response and in the components, and did not want to add extra code just so that I could rename one key")